frontend/udp/bytepool/bytepool.go

```diff
@@ -0,0 +1,34 @@
+package bytepool
+
+import "sync"
+
+// BytePool is a cached pool of reusable byte slices.
+type BytePool struct {
+	sync.Pool
+}
+
+// New allocates a new BytePool with slices of the provided capacity.
+func New(length, capacity int) *BytePool {
```

Comment:
Tbh I think we should only specify length. If people use different values for `length` and `capacity` (let's say 1024 and 4096) the length of slices that are pooled will be different from slices that are newly allocated because we reslice them to `[:cap(b)]` when we return them to the pool.